File: 0090-subsets-ii/0090-subsets-ii.go
func rec(i int, n int, nums []int, temp []int, answer *[][]int){
    if i == n {
        var exists bool = slices.ContainsFunc(*answer, func(a []int) bool { 
            return slices.Equal(a, temp)
        }) 
        if !exists {
            dup := make([]int,len(temp))
            copy(dup,temp)
            *answer = append(*answer, dup)
        }
        return
    }
    temp = append(temp,nums[i])
    rec(i+1,n,nums,temp,answer)
    temp = temp[:len(temp)-1]
    rec(i+1,n,nums,temp,answer)
}

func subsetsWithDup(nums []int) [][]int {
    sort.Ints(nums[:])
    var temp []int
    var answer [][]int
    rec(0,len(nums),nums,temp,&answer)
    return answer
}

File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    private void rec(int i, int n, int[] nums, List<Integer> temp, List<List<Integer>>answer){
        if(i == n){
            if(answer.indexOf(temp) == -1) answer.add(new ArrayList<>(temp));
            return;
        }
        temp.add(nums[i]);
        this.rec(i+1,n,nums,temp,answer);
        temp.removeLast();
        this.rec(i+1,n,nums,temp,answer);
    }

    public List<List<Integer>> subsetsWithDup(int[] nums) {
        int n = nums.length;
        Arrays.sort(nums);
        List<List<Integer>> answer = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        this.rec(0,n,nums,temp,answer);
        return answer;
    }
}